airflow/providers/slack/transfers/sql_to_slack.py

```diff
@@ -134,6 +141,8 @@ def execute(self, context: Context) -> None:
             output_file_name = fp.name
             output_file_format = output_file_format.upper()
             df_result = self._get_query_results()
+            if not self.allow_null and df_result.empty:
+                raise SqlToSlackNullOutputException("Sql output is empty.")
```

Comment:
Raising exception cause the task to fail. Instead of that, can we skip sending the slack message and keep the task status success?

Reply:
@dirrao updated, can you take another look? Thanks!

Reply:
@dirrao updated, can you take another look? Thanks!

Reply:
also @eladkal curious about your thoughts on the behavior of skipping sending the slack message instead of raising an Exception to fail the task.

Reply:
I think it makes sense but in this case probably we need another rename. Something like `set_skip_on_empty_result`

Reply:
If we would like to have multiple options we might make it as enum or literal string because action non-binary.
I would prefer literal, it just a string which wouldn't have any problem with serde.

```python
from typing_extensions import Literal

...
action_on_empty_df: Literal["send", "skip", "error", "ignore"] = "send"
```

- `send` - send even empty
- `skip` - raise AirflowSkipException, and task will skiped
- `error` - raise an error
- `ignore` - do nothing

Reply:
Naming of literal values could be different

Reply:
`action_on_empty_df: Literal["send", "skip", "error", "ignore"] = "send"`

That mixes between what what to do with the empty file and what state to set the task.